modules/core/shared/src/test/scala/gem/arb/ArbEphemerisKey.scala

```diff
@@ -28,7 +31,26 @@ trait ArbEphemerisKey {
     }
 
   implicit val CogenEphemerisKey: Cogen[EphemerisKey] =
-    Cogen[String].contramap(_.format)
+    Cogen[String].contramap(EphemerisKey.fromString.reverseGet)
+
+  private val perturbations: List[String => Gen[String]] =
+    List(
+      s => arbitrary[String],             // swap for a random string
+      s => Gen.const(s.replace("2", "0")) // create a leading zero, perhaps
```

Comment:
Why this?

Reply:
For testing we want to be sure to generate ephemeris keys that parse but *don't* format the same. So `"MajorBody_0123"` parses to `MajorBody(123)` which formats as `"MajorBody_123"`. By randomly replacing `2` with `0` we generate enough leading zeroes to test this. Otherwise the test fails because one of the laws checks that we're generating good enough test data.